Tasks/Npm/npmtask.ts

```diff
@@ -87,6 +97,16 @@ async function runNpmAuthHelperAsync(npmAuthRunner: trm.ToolRunner) : Promise<nu
     }
 }
 
+function getUserNpmrcPath() {
+    var userNpmRc = process.env['npm_config_userconfig'];
+    if(!userNpmRc){
+        // default npm rc is located at user's home folder.
+        userNpmRc = path.join(process.env['HOMEDRIVE'], process.env['HOMEPATH'], '.npmrc');
```

Comment:
For Linux/mac, I think we should only copy and only set the env var right before running auth helper.  that way we don't keep this broken on Linux.